src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java

```diff
@@ -56,15 +56,19 @@ private void deleteObjectsInBucket(String dataBucket) {
     try {
       Iterable<Blob> blobs = storage.list(dataBucket).iterateAll();
       for (Blob blob : blobs) {
-        boolean suceeded = storage.delete(blob.getBlobId());
-        if (suceeded) {
-          log.info("Deleted object {}", blob);
-        } else {
-          log.warn("Could not delete object {}", blob);
+        try {
+          boolean suceeded = storage.delete(blob.getBlobId());
+          if (suceeded) {
+            log.info("Deleted object {}", blob);
+          } else {
+            log.warn("Could not delete object {}", blob);
+          }
+        } catch (Exception e) {
```

Comment:
So following on from @patduin's earlier comment, is there a more specific exception type we can catch or do the Google APIs not really do checked exceptions?

Reply:
StorageExceptions are thrown but then I can't mock them in the tests as its a final class and I don't want to bring in PowerMock

Reply:
why do you need to mock them in the tests?